tests/test_md.py

```diff
@@ -661,7 +661,7 @@ def build_context(barostat_interval):
             weak_refs = []
             bps = []
             for p, bp in zip(sys_params, unbound_potentials):
-                bound_impl = bp.bind(p).bound_impl(np.float32)
+                bound_impl = bp.bind(p).to_gpu(np.float32).bound_impl
```

Comment:
Small surprise, not immediately clarified from inspecting `to_gpu` definition -- does `to_gpu(precision)` eagerly instantiate both bound and unbound impls? (so that the `.bound_impl` attribute is populated / ready to be referenced in this way?)

Reply:
> does to_gpu(precision) eagerly instantiate both bound and unbound impls?

No; there are two different methods: `Potential.to_gpu` and `BoundPotential.to_gpu` (the `.bind(p)` call turns a Potential into a BoundPotential. But the naming here seems confusing (using `bp` to refer to an (unbound) Potential).

Stepping back, it seems like there might be a mixed convention where "bound potential" sometimes refers to (1) a potential (reference or GPU) that has been partially applied to parameters, and sometimes to (2) a GPU implementation of a potential, whether or not it's an instance of `Potential` or `BoundPotential`. In this PR I changed some of the cases of (2) to (1), but there is still more work to do to make it consistent (first step is probably to agree on a convention).

Reply:
Thanks for clarifying! This diagnosis is helpful.

Regarding (2), I think previously we could use `bp.bound_impl()` or `bp.unbound_impl()`, but now bound potentials only support `bp.bound_impl`, unbound potentials only support `ubp.unbound_impl`, if I'm understanding correctly.

I think that makes sense as-is, but in a future refinement we might want to consider
(A) adding a method to `.unbind()` (`ubp = bp.unbind()`, so that the previous behavior can be reproduced `bp.to_gpu(np.float32).bound_impl`, `bp.unbind().to_gpu(np.float32).unbound_impl`,
(B) revising both names to `.impl`?

Reply:
> (A) adding a method to .unbind() (ubp = bp.unbind(), so that the previous behavior can be reproduced

The way to achieve this currently is `bp.potential` (i.e. just accessing the potential attribute). E.g. `bp.potential.to_gpu(np.float32).unbound_impl`

Reply:
Ahh, got it!

Reply:
Sorry, missed second part of response:

>I considered this, but overloading impl seemed a bit less explicit than what we had been doing, so didn't seem like an improvement. (Maybe I'm missing an argument as to why this would be preferable?)

No argument, just a reflex against repetition `bound_potential.bound_impl`, `unbound_potential.unbound_impl`.

On further thought, I agree with preference for current `.bound_impl`, `.unbound_impl` over `.impl` since it is more explicit.